plugins/explore/src/plugin.cpp

```diff
@@ -0,0 +1,244 @@
+        return screen.ToString() + '\n';
+      }
+
+    private:
+      printer_args args_;
+      ui_state state_;
+      std::unordered_map<type, ftxui::Component> components_;
+    };
+    return std::make_unique<screen_printer>(args_);
+  }
+
+  auto allows_joining() const -> bool override {
+    return true;
+  }
+
+  friend auto inspect(auto& f, table_printer& x) -> bool {
+    return f.object(x)
+      .pretty_name("table_printer")
+      .fields(f.field("args", x.args_));
+  }
+
+private:
+  printer_args args_;
+};
+
+class explore_plugin final : public virtual operator_plugin<explore_operator>,
+                             public virtual printer_plugin<table_printer> {
+public:
+  auto name() const -> std::string override {
+    return "explore";
+  }
+
+  auto parse_operator(parser_interface& p) const -> operator_ptr override {
+    auto parser
+      = argument_parser{"explore", fmt::format("https://docs.tenzir.com/docs/"
+                                               "connectors/sinks/explore")};
+    auto args = operator_args{};
+    parser.add("-f,--fullscreen", args.fullscreen);
+    parser.add("-w,--width", args.width, "<int>");
+    parser.add("-h,--height", args.height, "<int>");
+    parser.add("-n,--navigator-position", args.navigator_position, "<string>");
+    parser.add("-N,--navigator", args.navigator_auto_hide);
+    parser.add("-T,--hide-types", args.hide_types);
+    parser.parse(p);
+    if (args.width && !args.height)
+      diagnostic::error("--width requires also setting --height")
+        .primary(args.width->source)
+        .throw_();
+    else if (args.height && !args.width)
+      diagnostic::error("--height requires also setting --width")
+        .primary(args.height->source)
+        .throw_();
+    if (args.navigator_position) {
+      const auto& x = args.navigator_position->inner;
+      if (!(x == "left" || x == "right" || x == "top" || x == "bottom"))
+        diagnostic::error("invalid --navigator value")
+          .primary(args.navigator_position->source)
+          .note("must be one of 'left|right|top|bottom'")
+          .throw_();
+    }
+    return std::make_unique<explore_operator>(std::move(args));
+  }
+
+  auto parse_printer(parser_interface& p) const
+    -> std::unique_ptr<plugin_printer> override {
+    auto parser
+      = argument_parser{"explore", fmt::format("https://docs.tenzir.com/docs/"
+                                               "formats/table")};
+    auto args = printer_args{};
+    parser.add("-w,--width", args.width, "<int>");
+    parser.add("-h,--height", args.height, "<int>");
+    parser.add("-r,--real-time", args.real_time);
+    parser.add("-T,--hide-types", args.hide_types);
+    parser.parse(p);
+    return std::make_unique<table_printer>(std::move(args));
+  }
+};
+
+} // namespace
+
+} // namespace tenzir::plugins::explore
+
+TENZIR_REGISTER_PLUGIN(tenzir::plugins::explore::explore_plugin)
+// TENZIR_REGISTER_PLUGIN(tenzir::plugins::explore::table_plugin)
```

Comment:
This should get fixed before we merge it. The table format also needs its own documentation page.

Reply:
I would need your CMake assistance for this. Could I ask you to provide a patch that makes it possible to create two separately-named plugins within the same (non-builtin) plugin project?

Reply:
I don't think that's easily possible. You would at the very least need to duplicate the CMakeLists.txt file and the entire logic in it.

Did you consider putting the shared code into a library on its own that we link against from both plugins?

Reply:
I did think about it, but boy that's a lot of boilerplate. If you think that's worth the squeeze, I'll restructure the layout accordingly.

Reply:
It's the easiest solution. For the builtins, we generate this CMake for files that contain the `TENZIR_REGISTER_PLUGIN` macro, but for plugins we have to call `TenzirRegisterPlugin` in CMake once per plugin we add. Changing the CMake logic to allow multiple plugins in C++ per registered plugin in CMake will add a lot of complexity in a place that's already very hard to maintain.

Reply:
Okay, then I'll create a common `tui` library that `explore` and `table` depend on.

How would I express that dependency?